perceiver_pytorch/queries.py

```diff
@@ -24,25 +24,54 @@ def __init__(
         num_frequency_bands: int = 64,
         frequency_base: float = 2.0,
         sine_only: bool = False,
+        precomputed_fourier: Optional[torch.Tensor] = None,
+        use_both_precomputed_and_generated_fourier: bool = False,
     ):
         """
         Learnable Query with some inbuilt randomness to help with ensembling
 
         Args:
             channel_dim: Channel dimension for the output of the network
             query_shape: The final shape of the query, generally, the (T, H, W) of the output
+            conv_layer: The type of convolutional layer to use, either 3d or 2d
+            max_frequency: Max frequency for the Fourier Features
+            num_frequency_bands: Number of frequency bands for the Fourier Features
+            frequency_base: Base frequency for the Fourier Features
+            sine_only: Whether to use only the sine Fourier features
+            precomputed_fourier: Fourier features to use instead of computing them here,
+                useful for having temporally consistent features from history timesteps to future predictions
+                These features will be concatenated directly to the query, so should be compatible, and made in the
+                same way as in encode_position
+            use_both_precomputed_and_generated_fourier: Whether to use both the generated Fourier features giving the relative
```

Comment:
I think the parameter `use_both_precomputed_and_generated_fourier` could be renamed to something like `generate_fourier_features`.

Then, if the user wants:

* just precomputed fourier features, then just set `precomputed_fourier`
* just generated fourier features, then set `generate_fourier_features=True`
* _both_, then set both
* no fourier features, then set neither

Reply:
Sounds good! Made the change